### PR TITLE
add browscap-js 2.x

### PR DESCRIPTION
**Pull request by mimmi20**

### CODE REVIEW

**Review by jaydiablo — CHANGES_REQUESTED** [review]
A couple of things with this one:

1) Could you add the parser to the Readme?
2) I tested this parser, and it doesn't parse any UAs, which seems to be because the `node_modules/browscap-json-cache-files` directory is missing in the parser's directory (after installing with the `prepare` command). I had to wipe the `node_modules` folder and the `npm-shrinkwrap.json` file to get npm to install that dependency.  I did peek at the `npm-shrinkwrap.json` file before doing that, and it had the `browscap-json-cache-files` underneath the `browscap-js` entry, rather than on the same level. I'm not sure why it didn't flatten that out.  That's actually how it is in this PR: https://github.com/diablomedia/useragent-parser-comparison/pull/2/files#diff-6ed242b10dc1fb2d92746eed2330db37R18

Here's how it looks after I wiped and re-installed:

```json
{
  "name": "browscap-js-2",
  "version": "1.0.0",
  "dependencies": {
    "browscap-js": {
      "version": "2.2.6027",
      "from": "browscap-js@>=2.0.0 <3.0.0",
      "resolved": "https://registry.npmjs.org/browscap-js/-/browscap-js-2.2.6027.tgz"
    },
    "browscap-js-cache": {
      "version": "1.2.0",
      "from": "browscap-js-cache@>=1.2.0 <2.0.0",
      "resolved": "https://registry.npmjs.org/browscap-js-cache/-/browscap-js-cache-1.2.0.tgz"
    },
    "browscap-json-cache-files": {
      "version": "1.2.6027",
      "from": "browscap-json-cache-files@>=1.2.6027 <2.0.0",
      "resolved": "https://registry.npmjs.org/browscap-json-cache-files/-/browscap-json-cache-files-1.2.6027.tgz"
    },
    "charenc": {
      "version": "0.0.2",
      "from": "charenc@>=0.0.1 <0.1.0",
      "resolved": "https://registry.npmjs.org/charenc/-/charenc-0.0.2.tgz"
    },
    "crypt": {
      "version": "0.0.2",
      "from": "crypt@>=0.0.1 <0.1.0",
      "resolved": "https://registry.npmjs.org/crypt/-/crypt-0.0.2.tgz"
    },
    "is-buffer": {
      "version": "1.1.6",
      "from": "is-buffer@>=1.1.1 <1.2.0",
      "resolved": "https://registry.npmjs.org/is-buffer/-/is-buffer-1.1.6.tgz"
    },
    "md5": {
      "version": "2.2.1",
      "from": "md5@>=2.0.0 <3.0.0",
      "resolved": "https://registry.npmjs.org/md5/-/md5-2.2.1.tgz"
    },
    "synchronous-promise": {
      "version": "1.0.18",
      "from": "synchronous-promise@>=1.0.18 <2.0.0",
      "resolved": "https://registry.npmjs.org/synchronous-promise/-/synchronous-promise-1.0.18.tgz"
    }
  }
}
```

What version of NPM are you using? I'm on `3.10.9`.

If it is due to NPM difference, I wonder if it's possible to make the `parse` command find the proper location for that directory, whether it's nested or flat.  I bet we can make it use a similar command that's used here: https://github.com/mimmi20/browscap-js/blob/master/browscap.js#L5

Actually, it's better just to remove that directory parameter and let `browscap-js` find its own cache files, as I've done in this commit (to `browscap-js-1`): https://github.com/diablomedia/useragent-parser-comparison/commit/2c912fbe691676b7c33f945dcc16d6dbb27aa90e

I tested that change on `browscap-js-2` with my `npm-shrinkwrap.json` (contents above) and yours (from this PR) and it ran fine with both.

**Comment by mimmi20** [comment]
On my Windows System I'm using Node 9.3 with NPM 5.5, on the Linux VM I'm using Node 9.4 with NPM 5.6.
During the installtion of the node modules it compained that the format of the `shrinkwrap` file changed.

File: parsers/browscap-js-2/parse [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env node
+
+var initStart = process.hrtime();
+var Browscap = require('browscap-js');
+var browscap = new Browscap(__dirname + '/node_modules/browscap-json-cache-files/sources/');
```
Comment (jaydiablo) [inline parsers/browscap-js-2/parse]: Should be able to remove the directory definition here altogether and let your JS lib find the files based on `require` resolution, which should work for all versions of NPM (doesn't matter where the dependency is installed).  See this commit for an example: https://github.com/diablomedia/useragent-parser-comparison/commit/2c912fbe691676b7c33f945dcc16d6dbb27aa90e

**Review by jaydiablo — CHANGES_REQUESTED** [review]

**Comment by jaydiablo** [comment]
Thanks @mimmi20!

**Pull request merged**